### ISSUE TITLE
Increased loading time on checkout page due to Mollie assets

### ISSUE THREAD

**Issue by prestarocket** [mollie/PrestaShop#408]
### Expected behavior
Only load assets (js & css) when needed.
Can we only load assets on cart & order controller ?

### Actual behavior

Currently, the module loads via the ActionFrontControllerSetMedia hook all the js and css resources of the module (more than 200Kb on all pages).

### Steps to reproduce the behavior
Every front page load mollie assets
### Environment
* PrestaShop version: 1.7.7.8 
* Mollie module version: 4.4.0 
* PHP version: 7.2

**Comment by mariuszsienkiewicz** [comment]
I also confirm this issue.
On our sites, loading the mollie.js file on each page takes over 700 ms, which is a lot.

**Comment by Saimis777** [comment]
Hi @prestarocket and @mariuszsienkiewicz ,

Thank you for your messages!

We're currently looking in to this issue, I will update you with the process ASAP!

Thank you for your patience!

--
Best regards,
Invertus Support team

**Comment by AnatoleRozan** [comment]
I also confirm this issue. Could you patch it quickly? Thanks.

**Comment by prestarocket** [comment]
For your information, I have applied this modification in the module :
```

/* modif Prestarocket */
    public function shouldLoadMollie()
    {
        $load = false;
        if(isset($this->context->controller) && ($this->context->controller instanceof CartController || $this->context->controller instanceof OrderController || $this->context->controller instanceof OrderConfirmationController)){
            $show = true;

        }
       
        return $load;
    }

    /**
     * @throws PrestaShopException
     */
    public function hookActionFrontControllerSetMedia()
    {
        $load = $this->shouldLoadMollie();
        if(!$load){
            return;
        }
...

```

**Comment by Saimis777** [comment]
Hello,

Thank you for the provided information.

Loading Js&css is needed for the module to recognize the controllers that are used by custom one page checkout modules, without it loading, the module would not recognize them, and could potentially cause even more issues, we're currently checking how this could be optimized or changed.

Will keep you posted with any progress.

--
Best regards,
Invertus Support team

**Comment by prestarocket** [comment]
Hello @Saimis777 , we don t use custom one page modules. So we can only load assets in checkout page & confirmation page ?

**Comment by Saimis777** [comment]
Hello @prestarocket ,

Yes, from the code you've provided it should load only in checkout page & confirmation page, and you should be able to use it, but after any update the changes will need to be re-added.

I would suggest also cache for js/css, and minify the unneeded js/css loading. 
![image](https://user-images.githubusercontent.com/75423142/135277906-02c63ad6-2046-4083-b4e7-1f273d5d4070.png)


--
Best regards,
Invertus Support team

**Comment by mariuszsienkiewicz** [comment]
Hi @Saimis777 

enabling CCC is always a good advice but the problem is mainly connected with the remote JS file.

Maybe there should be an option to  disable the compatibility with third-party modules (custom one page checkouts etc.) and because of that we will not load remote JS files when it is not necessary.

**Comment by Saimis777** [comment]
@mariuszsienkiewicz ,

The team is looking at the options of what can be done.

As soon as we will have some answers/options/solutions, I will let you know!

--
Best regards,
Invertus Support team

**Comment by Saimis777** [comment]
Hello,

This raised issue is noted in the Mollie-Invertus board, once there will be updates regarding it, I will let you all know.

Closing the issue for now.

--
Best regards,
Invertus Support team

**Comment by loonas** [comment]
Hi @prestarocket @mariuszsienkiewicz @AnatoleRozan,

I hope that you are doing well!
I wanted to inform you that this issue has been fixed with our latest 5.0.0-beta2 version. You can find it and check it out [here](https://github.com/mollie/PrestaShop1.7/releases/tag/v5.0.0-beta2). 

Let us know if you have any questions or feedback!

Best regards,
Invertus Support team
